### PR TITLE
Refactor/convert etherscan into service

### PR DESCRIPTION
**Pull request by raiseandfall**
- refactor: Etherscan now a service like the Bitcoin ones
- refactor: update Explorer & Etherscan to use async/await
- test: added test for Etherscan

### CODE REVIEW

File: src/explorers/ethereum/etherscan.js [hunk 1]
```diff
@@ -1,44 +1,26 @@
...
-          // don't need to wrap this exception
-          reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-        }
-      }).catch(function () {
-        reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-      });
-  });
-  return etherScanFetcher;
-}
+const ETHERSCAN_API_KEY = 'FJ3CZWH8PQBV8W5U6JR8TMKAYDHBKQ3B1D';
```
Comment (lemoustachiste) [inline src/explorers/ethereum/etherscan.js:9]: we need to account somewhere, and I think that's what we forgot in #471, that you could pass your own keys for the explorers that are defined by default. It's not related to this PR, but we should not forget

File: src/explorers/ethereum/etherscan.js [hunk 2]
```diff
@@ -51,66 +33,57 @@ function parseEtherScanResponse (jsonResponse, block) {
...
-          // don't need to wrap this exception
-          reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-        }
-      }).catch(function () {
-        reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-      });
+async function checkEtherScanConfirmations (chain, blockNumber) {
+  const requestUrl = buildTransactionServiceUrl({
+    serviceUrls: getBlockNumberServiceUrls,
+    testApi: isTestChain(chain)
```
Comment (lemoustachiste) [inline src/explorers/ethereum/etherscan.js:61]: I think the `testAPI` property should better reflect that it's a boolean. `isTestAPI`?

File: test/application/certificate/verify-v3.test.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ describe('Certificate test suite', function () {
         let certificate;
 
         beforeEach(async function () {
-          sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+          sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/application/certificate/verify-v3.test.js:14]: this was Ethereum before, why the change to the bitcoin one?

File: test/application/certificate/verify-v3.test.js [hunk 4]
```diff
@@ -54,7 +54,7 @@ describe('Certificate test suite', function () {
...
```
Comment (lemoustachiste) [inline test/application/certificate/verify-v3.test.js:57]: same

File: test/application/certificate/verify-v3.test.js [hunk 5]
```diff
@@ -87,7 +87,7 @@ describe('Certificate test suite', function () {
...
```
Comment (lemoustachiste) [inline test/application/certificate/verify-v3.test.js:90]: same

File: test/e2e/verifier/ethereum-main-invalid-merkle-root.test.js [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-import * as ethereumExplorer from '../../../src/explorers/ethereum/etherscan';
+import * as Explorers from '../../../src/explorers/explorer';
 
 describe('given the certificate is an ethereum main with an invalid merkle root', function () {
   let certificate;
   let result;
 
   beforeAll(async function () {
-    sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+    sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-main-invalid-merkle-root.test.js:11]: Ethereum >< Bitcoin

File: test/e2e/verifier/ethereum-main-revoked.test.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+import * as Explorers from '../../../src/explorers/explorer';
 import sinon from 'sinon';
 
 describe('given the certificate is a revoked ethereum main', function () {
   let certificate;
   let result;
 
   beforeAll(async function () {
-    sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+    sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-main-revoked.test.js:11]: ETH >< BTC

File: test/e2e/verifier/ethereum-main-v2-valid.test.js [hunk 8]
```diff
@@ -1,11 +1,11 @@
 import { Certificate, VERIFICATION_STATUSES } from '../../../src';
 import FIXTURES from '../../fixtures';
-import * as ethereumExplorer from '../../../src/explorers/ethereum/etherscan';
+import * as Explorers from '../../../src/explorers/explorer';
 import sinon from 'sinon';
 
 describe('given the certificate is a valid ethereum main', function () {
   it('should verify successfully', async function () {
-    sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+    sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-main-v2-valid.test.js:8]: ETH >< BTC

File: test/e2e/verifier/ethereum-ropsten-revoked-no-revocationlist-2.0.test.js [hunk 9]
```diff
@@ -9,7 +9,7 @@ describe('given the certificate is a revoked certificate', function () {
     let result;
 
     beforeAll(async function () {
-      sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+      sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-ropsten-revoked-no-revocationlist-2.0.test.js:12]: ETH >< BTC

File: test/e2e/verifier/ethereum-ropsten-v2-valid.test.js [hunk 10]
```diff
@@ -1,11 +1,11 @@
 import { Certificate, VERIFICATION_STATUSES } from '../../../src';
 import FIXTURES from '../../fixtures';
-import * as ethereumExplorer from '../../../src/explorers/ethereum/etherscan';
+import * as Explorers from '../../../src/explorers/explorer';
 import sinon from 'sinon';
 
 describe('given the certificate is a valid ethereum ropsten', function () {
   it('should verify successfully', async function () {
-    sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+    sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-ropsten-v2-valid.test.js:8]: ETH >< BTC

File: test/e2e/verifier/ethereum-tampered.test.js [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
+import * as Explorers from '../../../src/explorers/explorer';
 import sinon from 'sinon';
 
 describe('given the certificate is a tampered ethereum', function () {
   let certificate;
   let result;
 
   beforeAll(async function () {
-    sinon.stub(ethereumExplorer, 'getEtherScanFetcher').resolves({
+    sinon.stub(Explorers, 'getBitcoinTransactionFromApi').resolves({
```
Comment (lemoustachiste) [inline test/e2e/verifier/ethereum-tampered.test.js:11]: ETH >< BTC

File: src/explorers/ethereum/etherscan.js (hunk 1 above)
Comment (raiseandfall) [reply src/explorers/ethereum/etherscan.js:9]: I first thought we would leave Etherscan parsing function in the explorers folder, so users could copy / paste it in their implementation. But we can open the API to send keys for the public explorer APIs we support. I'll come up with a model.

File: test/application/certificate/verify-v3.test.js (hunk 3 above)
Comment (raiseandfall) [reply test/application/certificate/verify-v3.test.js:14]: That's weird, I don't remember changing that file at all. I'll update.

File: test/application/certificate/verify-v3.test.js (hunk 4 above)
Comment (raiseandfall) [reply test/application/certificate/verify-v3.test.js:57]: Same too

File: test/application/certificate/verify-v3.test.js (hunk 5 above)
Comment (raiseandfall) [reply test/application/certificate/verify-v3.test.js:90]: Ok now that's weird

File: test/e2e/verifier/ethereum-main-invalid-merkle-root.test.js (hunk 6 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-main-invalid-merkle-root.test.js:11]: Even weirder

File: test/e2e/verifier/ethereum-main-revoked.test.js (hunk 7 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-main-revoked.test.js:11]: And yet still very weird

File: test/e2e/verifier/ethereum-main-v2-valid.test.js (hunk 8 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-main-v2-valid.test.js:8]: When is this ending

File: test/e2e/verifier/ethereum-ropsten-revoked-no-revocationlist-2.0.test.js (hunk 9 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-ropsten-revoked-no-revocationlist-2.0.test.js:12]: I'm confused

File: test/e2e/verifier/ethereum-ropsten-v2-valid.test.js (hunk 10 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-ropsten-v2-valid.test.js:8]: Oh well

File: test/e2e/verifier/ethereum-tampered.test.js (hunk 11 above)
Comment (raiseandfall) [reply test/e2e/verifier/ethereum-tampered.test.js:11]: Ah! final one. I suspect my editor did something weird there (auto replace or something).

File: src/explorers/ethereum/etherscan.js (hunk 1 above)
Comment (lemoustachiste) [reply src/explorers/ethereum/etherscan.js:9]: I think let's wait until we have the injection working. Things might be clearer then. But I agree, the explorer function should remain in the repo.

File: src/explorers/ethereum/etherscan.js (hunk 2 above)
Comment (raiseandfall) [reply src/explorers/ethereum/etherscan.js:61]: 👍

File: src/explorers/ethereum/etherscan.js (hunk 1 above)
Comment (raiseandfall) [reply src/explorers/ethereum/etherscan.js:9]: I agree.
